android/java/org/chromium/chrome/browser/rewards/BraveRewardsPanel.java

```diff
@@ -577,13 +571,7 @@ private boolean isValidNotificationType(int type, int argsNum) {
     }
 
     private void showNotification(String id, int type, long timestamp, String[] args) {
-        if (mBraveRewardsNativeWorker == null) {
-            return;
-        }
-
-        if (mExternalWallet == null
-                || (mExternalWallet.getStatus() == WalletStatus.NOT_CONNECTED
-                        && type == BraveRewardsNativeWorker.REWARDS_NOTIFICATION_GRANT)) {
+        if (mBraveRewardsNativeWorker == null || mExternalWallet == null) {
```

Comment:
Will this hide notifications if the user is unconnected (in which case `mExternalWallet` will be null)? If so, is that the intended behavior?

Reply:
if externalwallet is null, we won't be able to get to the wallet status anyways. and i don't think we need to show any notifications to the user if the user is in unconnected state. CC @Miyayes